create_publication.py

```diff
@@ -12,65 +12,55 @@
+from src.data_ingestion.get_data import download_zip_from_url
+from src.data_ingestion.reading_data import load_csv_into_spark_data_frame
+from src.processing.aggregate_counts import get_aggregate_counts, get_grouped_aggregate_counts
+from src.data_exports.write_csv import save_spark_dataframe_as_csv, rename_csv_output
+
 
 logger = logging.getLogger(__name__)
 
 def main():
     
     # load config, here we load our project's parameters from the config.toml file
     config = get_config("config.toml") 
-    server = config ['server']
-    database = config['database']
-    schema = config['schema']
-    table = config['table']
-    filled_value = config['filled_value']
-    output_dir = Path(config['output_dir'])
-    log_dir = Path(config['log_dir'])
+    # filled_value = config['filled_value']
+    # output_dir = Path(config['output_dir'])
+    # log_dir = Path(config['log_dir'])
+    # data_url = config['data_url']
+    # data_hes_path = config['hes_data_path']
 
     # configure logging
-    configure_logging(log_dir, config)
-    logger.info(f"Configured logging with log folder: {log_dir}.")
-
-    # sets up database connection
-    conn = make_database_connection(server, database)
+    configure_logging(config['log_dir'], config)
+    logger.info(f"Configured logging with log folder: {config['log_dir']}.")
 
-    # load data, this part handles importing our data sources     
-    query = read_sql_file('sql', 'example.sql', database, schema, table)
-    gp_df = get_df_from_server(conn, server, database, query)
+    # get artificial HES data as CSV
+    download_zip_from_url(config['data_url'], overwrite=True)
+    
+    # create spark session
+    spark = create_spark_session(config['project_name'])
 
-    # follow pre-processing steps  
-    gp_df.rename(columns={'ADDRESS_LINE_5': 'REGION', 
-                       'OPEN_DATE': 'OPENED', 
-                       'CLOSE_DATE': 'CLOSED'}, inplace=True)
+    # Loading data from CSV as spark data frame
+    df_hes_data = load_csv_into_spark_data_frame(spark, config['path_to_downloaded_data'])
     
-    gp_df = process_columns(gp_df, 
-        date_col_names = ['OPENED', 'CLOSED'], 
-        string_col_names= ['REGION', 'NAME']
-        )
+    # Creating dictionary to hold outputs
+    outputs = {}
 
-    gp_df = calculate_years(filled_value, gp_df)
-        
-    # prepare data for CSV
-    publication_breakdowns = {}
-    publication_breakdowns['gp_data'] = gp_df
+    # Count number of episodes in England - place this in the outputs dictionary
+    outputs["df_hes_england_count"] = get_aggregate_counts(df_hes_data, 'epikey', 'number_of_episodes')
 
-    # follow data processing steps
-    region_df = gp_count_by_region(gp_df)
-    region_df = calculate_mean_years(region_df, gp_df)
+    # Rename and save spark dataframes as CSVs:
+    for output_name in outputs:
+        save_spark_dataframe_as_csv(outputs[output_name], output_name)
+        rename_csv_output(output_name)
```

Comment:
This could use .items()
```suggestion
    for output_name, output in outputs.items():
        save_spark_dataframe_as_csv(output, output_name)
        rename_csv_output(output_name)
```